### PR TITLE
feat: add notifications onboarding wizard

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR implements a new step on the onboarding wizard to teach users about notifications feature. 

## **Related issues**

N/A

## **Manual testing steps**

Install a fresh app and accept the tour at beginning. 

## **Screenshots/Recordings**
https://github.com/MetaMask/metamask-mobile/assets/44679989/ff6273cb-613c-4b4d-afe8-99fb63d5445f

### **After**
Screenshot 2024-04-16 at 17 37 21

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Navbar/index.js [hunk 1]
```diff
@@ -981,14 +986,22 @@ export function getWalletNavbarOptions(
...
-      <ButtonIcon
-        variant={ButtonIconVariants.Primary}
-        onPress={openQRScanner}
-        iconName={IconName.Scan}
-        style={styles.infoButton}
-        size={IconSize.Xl}
-        testID={WalletViewSelectorsIDs.WALLET_SCAN_BUTTON}
-      />
+      <View style={styles.leftButtonContainer}>
+        <ButtonIcon
```
Comment (Cal-L) [inline app/components/UI/Navbar/index.js:990]: Adding note here that the button will be non-functional until the business logic is hooked up in follow up PRs.

**Review by Cal-L — COMMENTED** [review]
Thanks for the screenshot. Could you also provide a video of you going through the entire onboarding wizard? Let's also resolve the invalid PR template as well as E2E test failures

File: app/components/UI/Navbar/index.js (hunk 1 above)
Comment (Jonathansoufer) [reply app/components/UI/Navbar/index.js:990]: Indeed.

**Comment by Jonathansoufer** [comment]
> Thanks for the screenshot. Could you also provide a video of you going through the entire onboarding wizard? Let's also resolve the invalid PR template as well as E2E test failures

added

**Review by Cal-L — COMMENTED** [review]
Some comments: 
- Comparing with the production version of the app, the layout in step 1 of 5 appears on the top of the screen, then jumps down.
- Step 5 of 5 doesn't seem to be appearing in the video

File: app/components/UI/OnboardingWizard/Step7/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { shallow } from 'enzyme';
```
Comment (Cal-L) [inline app/components/UI/OnboardingWizard/Step7/index.test.tsx:2]: To resolve the fitness function check (prevents new use of enzyme), use `@testing-library/react-native` for unit tests. You can refer to other files that test using it.

Comment (Jonathansoufer) [reply app/components/UI/OnboardingWizard/Step7/index.test.tsx]: Fixed.

File: app/components/UI/OnboardingWizard/Step7/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import Step7 from '.';
+import configureMockStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+
+const mockStore = configureMockStore();
+const store = mockStore({});
+
+describe('Step7', () => {
+  it('should render correctly', () => {
+    const { toJSON } = render(
```
Comment (Cal-L) [inline app/components/UI/OnboardingWizard/Step7/index.test.tsx:12]: Use `renderWithProvider` instead. You can reference other usages in the app

Comment (Jonathansoufer) [reply app/components/UI/OnboardingWizard/Step7/index.test.tsx]: refactored

**Comment by Jonathansoufer** [comment]
> Some comments:
> 
> * Comparing with the production version of the app, the layout in step 1 of 5 appears on the top of the screen, then jumps down.
> * Step 5 of 5 doesn't seem to be appearing in the video

Fixed.

https://github.com/MetaMask/metamask-mobile/assets/44679989/085ee786-06e2-4ca2-83da-a6034a1866b5

**Review by georgewrmarshall — COMMENTED** [review]
Can approve after @Cal-L and @MetaMask/mobile-devs approval 👍

**Comment by Jonathansoufer** [comment]
> Can approve after @Cal-L and @MetaMask/mobile-devs approval 👍

@Cal-L is OoO. @sethkfman Could you take a look on this PR, please?

**Review by brianacnguyen — APPROVED** [review]

**Pull request merged**